agents/processor/src/processor.rs

```diff
@@ -233,45 +235,46 @@ impl Replica {
         Ok(Flow::Advance)
     }
 
-    #[instrument(err, level = "trace", skip(self), fields(self = %self))]
+    #[instrument(err, level = "info", skip(self), fields(self = %self, domain = message.message.destination, nonce = message.message.nonce, leaf_index = message.leaf_index, leaf = ?message.message.to_leaf()))]
     /// Dispatch a message for processing. If the message is already proven, process only.
     async fn process(&self, message: CommittedMessage, proof: NomadProof) -> Result<()> {
         use nomad_core::Replica;
+
+        // First check locally to see if we've tried before
+        if self.db.previously_attempted(&message)? {
+            info!("Message already attempted");
+            return Ok(());
+        }
+        // immediately store that we've attempted processing
+        self.db.set_previously_attempted(&message)?;
```

Comment:
What if process fails for RPC error? Then we have to manually restart and delete the db.

Reply:
Generally having the action of this loop as much determined by message status as possible was beneficial (resilient to unexpected errors). Maybe we can set to attempted IFF we have succeeded on the `match status { ... }` branch or if we match on the error and see something related to a revert. I know error string match is ugly but not sure how else to do this atm

Reply:
there's no correct response from the rpc that would cause us to _not_ want to mark this attempted

So the argument is that it minimizes edge case where _that precise rpc call_ fails and we have to then manually process?

Reply:
I'm thinking about it in reverse as in there _is_ (generally) a response from the RPC where we _do_ want to mark the message attempted even though the call failed.

If we mark all the calls that failed due to unreliable RPC errors processed, we will miss a lot of messages. Only marking them processed on success OR failure related to reverting, we definitely minimize the error case. I would not call it edge case though as long as we submit locally without tx manager since errors happens quite frequently

Reply:
> failure related to reverting

so the problem is that this is difficult to diagnose. We have a rough time determining which RPC errors are due to tx reverting and which are due to RPC issues. The ` provider::send_transaction` call we're using returns an `<M as Middleware>::Error`, which gets type-erased and boxed in our `ChainCommunicationError::MiddlewareError` type. So to determine whether it's a revert or something else, we'd have to attempt to upcast it and then trawl through it to figure out whether it's an RPC issue or a revert

So probably the best we can do is 
- check attempted (and shortcut out)
- check on-chain status
- mark attempted
- dispatch tx